### PR TITLE
String range implemented according to #4.

### PR DESCRIPTION
**Pull request by anev**
String range implemented according to #52.
Now `{{ theString[1:4] }}` will render as a substring.

More examples:
```
{{ theString[1:] }}
{{ theString[:3] }}
{{ theString[-7:-4] }}
```

### CODE REVIEW

File: src/main/java/com/hubspot/jinjava/el/ext/AstRangeBracket.java [hunk 1]
```diff
@@ -1,21 +1,19 @@
...
-import javax.el.PropertyNotFoundException;
-
 import com.hubspot.jinjava.objects.collections.PyList;
-
 import de.odysseus.el.misc.LocalMessages;
 import de.odysseus.el.tree.Bindings;
 import de.odysseus.el.tree.impl.ast.AstBracket;
 import de.odysseus.el.tree.impl.ast.AstNode;
 
+import javax.el.ELContext;
```
Comment (boulter) [inline src/main/java/com/hubspot/jinjava/el/ext/AstRangeBracket.java]: could you preserve the import order here?

File: src/main/java/com/hubspot/jinjava/el/ext/AstRangeBracket.java [hunk 2]
```diff
@@ -81,6 +86,19 @@ public Object eval(Bindings bindings, ELContext context) {
     return result;
   }
 
+  private String evalString(String base, int startNum, int endNum) {
+    if (startNum > endNum) {
+      return "";
+    }
+    if (startNum <= 0) {
```
Comment (boulter) [inline src/main/java/com/hubspot/jinjava/el/ext/AstRangeBracket.java]: could replace these with `Math.max` and `Math.min`

File: src/test/java/com/hubspot/jinjava/tree/ExpressionNodeTest.java [hunk 3]
```diff
@@ -1,17 +1,16 @@
...
-
-import org.junit.Before;
-import org.junit.Test;
-
 import com.google.common.base.Throwables;
 import com.google.common.io.Resources;
 import com.hubspot.jinjava.Jinjava;
 import com.hubspot.jinjava.interpret.Context;
 import com.hubspot.jinjava.interpret.JinjavaInterpreter;
+import org.junit.Before;
```
Comment (boulter) [inline src/test/java/com/hubspot/jinjava/tree/ExpressionNodeTest.java]: please preserve import order

File: src/test/java/com/hubspot/jinjava/el/ext/RangeStringTest.java [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+        assertThat(jinjava.render("{{ theString[3:8] }}", context)).isEqualTo("ti\nli");
+    }
+
+    @Test
+    public void testStringRangeCyrillic() {
+        Map<String, Object> context = new HashMap<>();
+        context.put("theString", "Строка с non ascii символами");
+
+        assertThat(jinjava.render("{{ theString[1:4] }}", context)).isEqualTo("тро");
+    }
```
Comment (boulter) [inline src/test/java/com/hubspot/jinjava/el/ext/RangeStringTest.java]: Could you also support negative and unspecified indexes such as `[-7:-3]` and `[3:]` for compatibility with python and Jinja? Some good examples here: http://pythoncentral.io/cutting-and-slicing-strings-in-python/

**Comment by boulter** [comment]
Thanks for your contribution!

File: src/main/java/com/hubspot/jinjava/el/ext/AstRangeBracket.java (hunk 1 above)
Comment (anev) [inline src/main/java/com/hubspot/jinjava/el/ext/AstRangeBracket.java]: @boulter 
Not sure which is right. Is the order correct?
1. java.*
2. javax.*
3. com.*
4. de.*

**Comment by anev** [comment]
@boulter 
empty ranges like `[:5]` are possible now, negative like `[-7:-2]` too.

Hope this pull request is useful for the project.

**Pull request merged**